records_mover/records/schema/field/sqlalchemy.py

```diff
@@ -124,75 +124,73 @@ def field_from_sqlalchemy_column(column: Column,
-               decimal_constraints.fixed_scale is not None):
-                return driver.type_for_fixed_point(precision=decimal_constraints.fixed_precision,
-                                                   scale=decimal_constraints.fixed_scale)
-
-            if (decimal_constraints.fp_total_bits is not None and
-               decimal_constraints.fp_significand_bits is not None):
-                return driver.\
-                    type_for_floating_point(fp_total_bits=decimal_constraints.fp_total_bits,
-                                            fp_significand_bits=decimal_constraints.
-                                            fp_significand_bits)
-
-        # default to an IEEE double absent more information:
-        #
-        # https://en.wikipedia.org/wiki/Double-precision_floating-point_format
-        return driver.type_for_floating_point(fp_total_bits=64,
-                                              fp_significand_bits=53)
-
-    elif field.field_type == 'boolean':
-        return sqlalchemy.sql.sqltypes.BOOLEAN()
-    elif field.field_type == 'string':
-        if field.constraints and\
-           not isinstance(field.constraints, RecordsSchemaFieldStringConstraints):
-            raise ValueError(f"Incorrect constraint type in {field.name}: {field.constraints}")
-
-        if field.statistics and\
-           not isinstance(field.statistics, RecordsSchemaFieldStringStatistics):
-            raise ValueError(f"Incorrect statistics type in {field.name}: {field.statistics}")
-
-        string_constraints =\
-            cast(Optional[RecordsSchemaFieldStringConstraints], field.constraints)
-        string_statistics =\
-            cast(Optional[RecordsSchemaFieldStringStatistics], field.statistics)
-        n = generate_string_length(string_constraints, string_statistics, driver)
-        return sqlalchemy.sql.sqltypes.String(n)
-    elif field.field_type == 'date':
-        return sqlalchemy.sql.sqltypes.DATE()
-    elif field.field_type == 'datetime' or field.field_type == 'datetimetz':
-        has_tz = field.field_type == 'datetimetz'
-        return driver.type_for_date_plus_time(has_tz=has_tz)
-    elif field.field_type == 'time' or field.field_type == 'timetz':
-        if driver.supports_time_type():
-            has_tz = field.field_type == 'timetz'
-            return sqlalchemy.sql.sqltypes.TIME(timezone=has_tz)
-        else:
-            # HH:MM AM; should be same in bytes as chars at least in UTF-8
-            return sqlalchemy.sql.sqltypes.VARCHAR(8)
-    else:
-        raise NotImplementedError("Teach me how to handle records schema "
-                                  f"type {field.field_type}")
+    field_to_func_map = {
+        'integer': get_integer_type,
+        'decimal': get_decimal_type,
+        'boolean': cast(Callable[['RecordsSchemaField', 'DBDriver'], sqlalchemy.types.TypeEngine],
+                        lambda field, driver: sqlalchemy.sql.sqltypes.BOOLEAN()),
```

Comment:
is there a way to `cast()` the entries in the map at the end all at once instead of each time individually?